meta-mbl-distro/scripts/lib/update/util.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import inspect
+import pathlib
+
+
+class ArchivedFileSpec:
+    """
+    A specification of a file to be added to an archive.
+
+    The class holds two pieces of information: the path to the file to be
+    added, and the path/filename that the file should have within the archive.
+    """
+
+    def __init__(self, path, archived_path=None):
+        """
+        Create a new ArchivedFileSpec.
+
+        :param path str|Path: path to the file to be archived. The path is
+            resolved so that if it points to a symlink, the underlying file
+            will be used rather than the symlink itself. The file must already
+            exist.
+        :param archived_path str|Path: path to the file within the archive. If
+            set to None (the default value) then the file will be added to the
+            archive with the same name as outside the archive (before symlink
+            resolution).
+        """
+        self.path = pathlib.Path(path)
+        if archived_path is None:
+            self.archived_path = path.name
+        else:
+            self.archived_path = pathlib.Path(archived_path)
+
+        try:
+            # We don't want symlinks in our payloads
+            self.path = self._strict_path_resolve(self.path)
+        except FileNotFoundError:
+            bb.fatal(
+                'The file "{}" was not found.'
+                " Have you built all of the required components?".format(
+                    self.path
+                )
+            )
+
+    @staticmethod
+    def _strict_path_resolve(path):
```

Comment:
> Should we make this a 'public' free function? I think we could re-use it in other scripts, and it wouldn't make much sense to have to refer to the ArchivedFileSpec if we wanted to use it in a different context.

Agree.

> Thinking about it, the read_chunks generator and get_bitbake_conf_var would be also pretty useful for other scripts.

I agree about `read_chunks`. I'm not sure I agree about `get_bitbake_conf_var`... It does the same thing as BitBake's `getVar` but with error handling that is less generally useful.

Let me see what I can do...

Reply:
I've moved `strict_path_resolve` and `read_chunks` into a fileutils lib. I've moved `get_bitbake_conf_var` into a tinfoilutils lib - I think it might be useful for other scripts that use Tinfoil, but not other BitBake stuff.

https://github.com/ARMmbed/meta-mbl/pull/763/commits/7553785672934548c33b8c6fda45f478b2855208